### PR TITLE
[TA] Improve documentation

### PR DESCRIPTION
**Pull request by maririos**
In general, it updates Readme and Samples docs/
- Uses `Analyze` instead of `Recognize` for Healthcare
- Removes information in docs (docstring will be later) about Healthcare in preview and AAD not working
- Removes redundancy on Healthcare samples and updates snippets and makes sure sync samples are actually sync and async are async, not a mix
- Updated Polling LRO sample
- Fixes the PII Categories sample
- Removes outdated opinion mining diagram.

### CODE REVIEW

File: sdk/textanalytics/Azure.AI.TextAnalytics/README.md [hunk 1]
```diff
@@ -159,7 +159,7 @@ The following section provides several code snippets using the `client` [created
 ### Async examples
 * [Detect Language Asynchronously](#detect-language-asynchronously)
 * [Recognize Entities Asyncronously](#recognize-entities-asynchronously)
-* [Recognize Healthcare Entities Asyncronously](#recognize-healthcare-entities-asynchronously)
+* [Analyze Healthcare Entities Asyncronously](#analyze-healthcare-entities-asynchronously)
```
Comment (kristapratico) [inline sdk/textanalytics/Azure.AI.TextAnalytics/README.md:162]: nit: spelling here and for the above line
```suggestion
* [Analyze Healthcare Entities Asynchronously](#analyze-healthcare-entities-asynchronously)
```

File: sdk/textanalytics/Azure.AI.TextAnalytics/README.md [hunk 2]
```diff
@@ -837,7 +825,7 @@ This project has adopted the [Microsoft Open Source Code of Conduct][code_of_con
 [custom_subdomain]: https://docs.microsoft.com/azure/cognitive-services/authentication#create-a-resource-with-a-custom-subdomain
 [DefaultAzureCredential]: https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/identity/Azure.Identity/README.md
 [logging]: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/Diagnostics.md
-[data_limits]: https://docs.microsoft.com/azure/cognitive-services/text-analytics/overview#data-limits
+[data_limits]: https://docs.microsoft.com/azure/cognitive-services/text-analytics/concepts/data-limits?tabs=version-3
```
Comment (kristapratico) [inline sdk/textanalytics/Azure.AI.TextAnalytics/README.md:828]: ohh nice, i'll update this too

**Review by kristapratico — APPROVED** [review]

File: sdk/textanalytics/Azure.AI.TextAnalytics/README.md (hunk 2 above)
Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/README.md:828]: the link still works in case you want to leave it. It just makes the user do an extra jump

Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/README.md:828]: Ohh you made me realize I have this link everywhere. Might as well update it everywhere too :P

File: sdk/textanalytics/Azure.AI.TextAnalytics/README.md (hunk 1 above)
Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/README.md]: thank you!

**Pull request merged**